cmake-tool/helpers/check_arch_compiler.cmake

```diff
@@ -101,7 +96,7 @@ function(check_arch_gcc)
 endfunction()
 
 function(check_arch_compiler)
-    if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    if(CMAKE_C_COMPILER_ID MATCHES "Clang")
```

Comment:
Please consider making this a separate commit with a comment that explains the issue that is fixed here.